packages/assets-controllers/src/NftController.ts

```diff
@@ -1218,14 +1269,23 @@ export class NftController extends BaseController<
     origin: string,
     {
       networkClientId,
-      userAddress = this.#selectedAddress,
+      userAddress,
     }: {
       networkClientId?: NetworkClientId;
       userAddress?: string;
-    } = {
-      userAddress: this.#selectedAddress,
-    },
+    } = {},
   ) {
+    const selectedAccount = this.messagingSystem.call(
+      'AccountsController:getAccount',
+      this.#selectedAccountId,
+    );
+
+    // Previously selectedAddress could be an empty string. This is to preserve the behaviour
+    userAddress = userAddress || selectedAccount?.address || '';
+    if (!userAddress) {
+      return;
+    }
```

Comment:
I feel like we could have a method for this too. For 2 things:

1. We repeat this pattern multiple times
2. We don't always need to call `:getAccount` (if `userAddres` is not empty)

So maybe something like:
```ts
#getAddressOrSelectedAddress(address: string | undefined): string {
  if (address) {
    return address;
  }
  
  // If the address is not defined (or empty), we fallback to the currently selected account's address
  const selectedAccount = this.messagingSystem.call(
    'AccountsController:getAccount',
    this.#selectedAccountId,
  );
  return selectedAccount?.address || '';
}
```

So we can use this one like:
```ts
const address = this.#getAddressOrSelectedAddress(userAddress);
if (!address) {
  return;
}
```

WDYT?